sdk/storage/azure-storage-blob-changefeed/src/main/java/com/azure/storage/blob/changefeed/BlobLazyDownloader.java

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.storage.blob.changefeed;
+
+import com.azure.storage.blob.BlobAsyncClient;
+import com.azure.storage.blob.implementation.util.ChunkedDownloadUtils;
+import com.azure.storage.blob.models.BlobDownloadAsyncResponse;
+import com.azure.storage.blob.models.BlobRange;
+import com.azure.storage.blob.models.BlobRequestConditions;
+import com.azure.storage.blob.models.ParallelTransferOptions;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+import reactor.core.scheduler.Schedulers;
+
+import java.nio.ByteBuffer;
+import java.util.function.Function;
+
+/**
+ * FOR INTERNAL USE ONLY.
+ * Class to lazily download a blob.
+ */
+class BlobLazyDownloader {
```

Comment:
This type doesn't seem like it's actually lazy to me. While it is sequential, it will still continue to issue download requests as long as the subscription to the data is not cancelled. I haven't seen yet where this is being used, but it will probably need to apply back pressure in order to actually slow down requests and make it lazy.